### ISSUE TITLE
More permissive version range for the @opentelemetry/instrumentation dependency

### ISSUE THREAD

**Issue by amakhrov** [gas-buddy/opentelemetry-instrumentation-fetch-node#18]
Currently it's defined as
```
"@opentelemetry/instrumentation": "^0.46.0",
```
Note that since it's pre-1.0.0 version, it means the range of `>=0.46.0 && <0.47.0`
This is already behind the current version of opentelemetry (`0.52.1` at this moment). It also misses some important (at least, to us) bug fixes from the latest opentelemetry version.

When used as a part of a larger solution (e.g. getsentry-js uses `opentelemetry` directly and also `opentelemetry-instrumentation-fetch-node` as an additional dep), it results in having two separate version of the opentelemetry in the project.

Is there a reason this package wouldn't work with the latest opentelemetry version?

**Comment by jd-carroll** [comment]
@timfish @mydea - I'm sure I already annoyed you enough earlier this year with ESM (which 🙏🏻 again). Would be great to get some updates here to allow for a more permissive range as @amakhrov has requested.

Note: I can't tell who actually owns this repo (or has commit priv)

**Comment by mydea** [comment]
Yeah, this is a general probem with `@opentelemetry/instrumentation` being pre-1 😬 Neither @timfish nor I maintain this package here, but it is an additional challenge for this package. IMHO the only thing that can be done for this package is to keep bumping the `@opentelemetry/instrumentation` dependency and releasing new versions every time 😬

**Comment by djMax** [comment]
I thought that the core team made a version based on this package but now "official." I can't find it from my phone here but  maybe it's called undici instead?

**Comment by timfish** [comment]
There's `@opentelemetry/instrumentation-undici` and I'm in the process of migrating Sentry's Node SDK to use this after some previous PR's to bring it's config to the same level.
